haystack/components/generators/hugging_face_api.py

```diff
@@ -242,7 +242,12 @@ def _stream_and_build_response(
                 first_chunk_time = datetime.now().isoformat()
 
             stream_chunk = StreamingChunk(
-                content=token.text, meta=chunk_metadata, component_info=component_info, index=0, start=len(chunks) == 0
+                content=token.text,
+                meta=chunk_metadata,
+                component_info=component_info,
+                index=0,
+                start=len(chunks) == 0,
+                finish_reason=chunk_metadata.get("finish_reason"),
```

Comment:
Same question here. When using the HuggingFaceAPI do you know what possible finish reasons we can have?